cl/alerts/views.py

```diff
@@ -60,16 +64,116 @@ def delete_alert_confirm(request, pk):
     )
 
 
-def disable_alert(request, secret_key):
+@csrf_exempt
+@require_http_methods(["POST"])
+@ratelimiter_unsafe_3_per_m
+def one_click_disable_alert(request: HttpRequest, secret_key: str):
     """Disable an alert based on a secret key."""
     alert = get_object_or_404(Alert, secret_key=secret_key)
-    prev_rate = alert.rate
     alert.rate = Alert.OFF
     alert.save()
+    # Mail clients send POSTs; an ugly response is OK.
+    return HttpResponse("You have been successfully unsubscribed!")
+
+
+@require_http_methods(["POST"])
+def htmx_disable_alert(request: HttpRequest, secret_key: str):
+    """Disables a specified alert within an HTMX-powered page."""
+    if not request.META.get("HTTP_HX_REQUEST"):
+        return HttpResponseNotFound(
+            "Your attempt to disable the alert was unsuccessful."
+        )
+
+    alert = get_object_or_404(Alert, secret_key=secret_key)
+    alert.rate = Alert.OFF
+    alert.save()
+    # Mail clients send POSTs; an ugly response is OK.
```

Comment:
This comment doesn't make sense here, does it?

Reply:
You're right, this comment doesn't make sense. I'll remove it.